src/Qir/Runtime/lib/Simulators/OpenSystemsSimulator.cpp

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <assert.h>
+#include <bitset>
+#include <complex>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include "QirRuntimeApi_I.hpp"
+#include "QSharpSimApi_I.hpp"
+#include "SimFactory.hpp"
+
+using namespace std;
+typedef long long int i64;
```

Comment:
What is the advantage of `i64` over the [`uint64_t`](https://man7.org/linux/man-pages/man0/stdint.h.0p.html)?

Reply:
None in particular; this type is used only for error return codes from C API calls, at which point it's fairly arbitrary whether we assign errors to positive or negative numbers. [In `c_api.rs`](https://github.com/microsoft/qsharp-runtime/blob/72977df6a9ddbc17b9dcecc3dfb7fd355f348a81/src/Simulation/qdk_sim_rs/src/c_api.rs#L41), I used `-1` as the error code, thinking of the convention used by Unix processes. That said, if you'd strongly suggest changing the return code convention to `u64` (Rust) / `uint64_t` (C / C++), I'm happy to do so.

Reply:
Sorry, the `u` was redundant in my question. I was trying to suggest `int64_t` instead of `i64`.
The way the `i64` is currently defined does not guarantee that `i64` is 64-bit type (can be larger than that). Whereas `int64_t` is strictly 64-bit type.